### ISSUE TITLE
try/except, if-else and no-else-return

### ISSUE THREAD

**Issue by mmmm1998** [PyCQA/pylint#2147]
```python3
def function(function1, function2, argument):
    try:
        result = function1(argument)
    except Exception:
        # some logging stuff here
        return None

    if result != 42:
        return function2(result)
    else:
        return 0
```
I think, this code shouldn't get flagged with `no-else-return`, but it does. I'am wrong?

### pylint --version output
No config file found, using default configuration
pylint 1.9.1, 
astroid 1.6.2
Python 3.6.5 (default, Apr  1 2018, 05:46:30) 
[GCC 7.3.0]

**Comment by mmmm1998** [comment]
I think, it possible related with #1852

**Comment by PCManticore** [comment]
The check is right. You can remove the `else` clause and deindent the `return 0` one level to the left.
